DigitalLearningSolutions.Data.Tests/Services/RegistrationServiceTests.cs

```diff
@@ -932,6 +938,181 @@ public void PromoteDelegateToAdmin_calls_data_service_with_expected_value()
+                userId,
+                "987.654.321.100",
+                false
+            );
+
+            // Then
+            A.CallTo(
+                    () =>
+                        registrationDataService.RegisterDelegateAccountAndCentreDetailForExistingUser(
+                            A<DelegateRegistrationModel>.That.Matches(d => !d.Approved),
+                            userId,
+                            currentTime,
+                            null
+                        )
+                )
+                .MustHaveHappened();
+            approved.Should().BeFalse();
+        }
+
+        [Test]
+        public void CreateDelegateAccountForExistingUser_sends_approval_email_with_old_site_approval_link()
+        {
+            // Given
+            const bool refactoredTrackingSystemEnabled = false;
+            const int userId = 2;
+            var model = RegistrationModelTestHelper.GetDefaultInternalDelegateRegistrationModel();
+
+            // When
+            registrationService.CreateDelegateAccountForExistingUser(
+                model,
+                userId,
+                string.Empty,
+                refactoredTrackingSystemEnabled
+            );
+
+            // Then
+            A.CallTo(
+                () =>
+                    emailService.SendEmail(
+                        A<Email>.That.Matches(
+                            e =>
+                                e.To[0] == ApproverEmail &&
+                                e.Cc.IsNullOrEmpty() &&
+                                e.Bcc.IsNullOrEmpty() &&
+                                e.Subject == "Digital Learning Solutions Registration Requires Approval" &&
+                                e.Body.TextBody.Contains(OldSystemBaseUrl + "/tracking/approvedelegates")
+                        )
+                    )
+            ).MustHaveHappened();
+        }
+
+        [Test]
+        public void CreateDelegateAccountForExistingUser_sends_approval_email_with_refactored_tracking_system_link()
+        {
+            // Given
+            const bool refactoredTrackingSystemEnabled = true;
+            const int userId = 2;
+            var model = RegistrationModelTestHelper.GetDefaultInternalDelegateRegistrationModel();
+
+            // When
+            registrationService.CreateDelegateAccountForExistingUser(
+                model,
+                userId,
+                string.Empty,
+                refactoredTrackingSystemEnabled
+            );
+
+            // Then
+            A.CallTo(
+                () =>
+                    emailService.SendEmail(
+                        A<Email>.That.Matches(
+                            e =>
+                                e.To[0] == ApproverEmail &&
+                                e.Cc.IsNullOrEmpty() &&
+                                e.Bcc.IsNullOrEmpty() &&
+                                e.Subject == "Digital Learning Solutions Registration Requires Approval" &&

```

Comment:
I'd be inclined to move these conditions into another test, and just check the approval link in these tests. It makes it easier to see what's actually being tested, and easier to debug if the test breaks as a result of one of these things is suddenly not true.